receiver/bindplaneauditlogs/logs.go

```diff
@@ -0,0 +1,236 @@
+
+package bindplaneauditlogs
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"sort"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+const (
+	bindplaneTimeFormat = time.RFC3339
+)
+
+// Add a wrapper struct for the API response
+type apiResponse struct {
+	AuditEvents []AuditLogEvent `json:"auditEvents"`
+}
+
+type bindplaneAuditLogsReceiver struct {
+	cfg           Config
+	client        *http.Client
+	consumer      consumer.Logs
+	logger        *zap.Logger
+	cancel        context.CancelFunc
+	wg            *sync.WaitGroup
+	lastTimestamp time.Time
+	settings      component.TelemetrySettings
+}
+
+// newBindplaneAuditLogsReceiver returns a newly configured bindplaneAuditLogsReceiver
+func newBindplaneAuditLogsReceiver(cfg *Config, logger *zap.Logger, consumer consumer.Logs) (*bindplaneAuditLogsReceiver, error) {
+	return &bindplaneAuditLogsReceiver{
+		cfg:           *cfg,
+		consumer:      consumer,
+		logger:        logger,
+		wg:            &sync.WaitGroup{},
+		lastTimestamp: time.Now().UTC().Add(-cfg.PollInterval),
+	}, nil
+}
+
+func (r *bindplaneAuditLogsReceiver) Start(ctx context.Context, host component.Host) error {
+	client, err := r.cfg.ToClient(ctx, host, r.settings)
+	if err != nil {
+		return fmt.Errorf("failed to create HTTP client: %w", err)
+	}
+	r.client = client
+
+	ctx, cancel := context.WithCancel(ctx)
+	r.cancel = cancel
+	r.wg.Add(1)
+	go r.startPolling(ctx)
+	return nil
+}
+
+func (r *bindplaneAuditLogsReceiver) startPolling(ctx context.Context) {
+	defer r.wg.Done()
+	t := time.NewTicker(r.cfg.PollInterval)
+
+	err := r.poll(ctx)
+	if err != nil {
+		r.logger.Error("there was an error during the first poll", zap.Error(err))
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-t.C:
+			err := r.poll(ctx)
+			if err != nil {
+				r.logger.Error("there was an error during the poll", zap.Error(err))
+			}
+		}
+	}
+}
+
+func (r *bindplaneAuditLogsReceiver) poll(ctx context.Context) error {
+	logEvents, err := r.getLogs(ctx)
+	if err != nil {
+		return err
+	}
+	observedTime := pcommon.NewTimestampFromTime(time.Now())
+	logs := r.processLogEvents(observedTime, logEvents)
+	if logs.LogRecordCount() > 0 {
+		if err := r.consumer.ConsumeLogs(ctx, logs); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *bindplaneAuditLogsReceiver) getLogs(ctx context.Context) ([]AuditLogEvent, error) {
+	var logs []AuditLogEvent
+	const timeout = 1 * time.Minute
+
+	reqURL := &url.URL{
+		Host: r.cfg.bindplaneURL.Host,
+		Path: "/v1/audit-events",
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, "GET", reqURL.String(), nil)
+	if err != nil {
+		err = fmt.Errorf("error creating request: %w", err)
+		return nil, err
+	}
```

Comment:
The errors in this function should be wrapped in the return statement. Also they should not have the word "error" or "failed" in the error message itself. The errors will get logged as an error so it'll be redundant and take up a lot of extra space in the log. As a general rule of thumb, the error msg should tell you what function produced the error. This is an example of how to wrap one of these errors.
```suggestion
	req, err := http.NewRequestWithContext(ctx, "GET", reqURL.String(), nil)
	if err != nil {
		return nil, return nil, fmt.Errorf("newRequestWithContext: %w", err)
	}
```